ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -323,20 +330,26 @@ def empty_action(self, n_agents: int) -> ActionTuple:
         for a number of agents.
         :param n_agents: The number of agents that will have actions generated
         """
-        continuous = np.zeros((n_agents, self.continuous_size), dtype=np.float32)
-        discrete = np.zeros((n_agents, self.discrete_size), dtype=np.int32)
-        return ActionTuple(continuous, discrete)
+        action_tuple = ActionTuple()
+        action_tuple.add_continuous(
+            np.zeros((n_agents, self.continuous_size), dtype=np.float32)
+        )
+        action_tuple.add_discrete(
+            np.zeros((n_agents, self.discrete_size), dtype=np.int32)
+        )
+        return action_tuple
 
     def random_action(self, n_agents: int) -> ActionTuple:
         """
         Generates ActionTuple corresponding to a random action (either discrete
         or continuous) for a number of agents.
         :param n_agents: The number of agents that will have actions generated
         """
+        action_tuple = ActionTuple()
         continuous = np.random.uniform(
             low=-1.0, high=1.0, size=(n_agents, self.continuous_size)
         )
-        discrete = np.zeros((n_agents, self.discrete_size), dtype=np.int32)
+        action_tuple.add_continuous(continuous)
```

Comment:
Can we just do something like 

```suggestion
        _continuous = continuous
        _discrete = np.column_stack() if self.discrete_size > 0 else None
        action_tuple = ActionTuple(_continuous, _discrete)
```